pkg/action/install.go

```diff
@@ -195,7 +197,10 @@ func (i *Install) Run(chrt *chart.Chart, vals map[string]interface{}) (*release.
 	if i.ClientOnly {
 		// Add mock objects in here so it doesn't use Kube API server
 		// NOTE(bacongobbler): used for `helm template`
-		i.cfg.Capabilities = chartutil.DefaultCapabilities
+		i.cfg.Capabilities = chartutil.DefaultCapabilities.Copy()
```

Comment:
In actual use, this doesn't really matter since the command exits immediately, but tests seem to reuse the same global instance so any changes here were affecting downstream tests.

Reply:
This could potentially affect users who are consuming the Go SDK and it could be argued that this could be considered a behavioural change.

What's the impact here?

Reply:
In my opinion, the existing behavior is a bug as it was leaking changes for `APIVersions` to unrelated calls. So instead of setting `APIVersions` only for the current call of `Run`, it affects all further calls to `Run` even with different sets of `APIVersions`. It would effectively just keep appending `APIVersions` every time without "resetting" it to the default set.

The code path itself is only for `ClientOnly` which rules out most helm usage. The two main use cases left are `helm template` which shouldn't see any change in behavior and anything that calls the Go SDK to generate manifests. For the latter case, there would only be a change of behavior if they call `Run` with differing `APIVersions` and the charts they're using use those `APIVersions` somehow. I believe the only users like this would be something like `argocd` that generates manifests to be applied separately against a running Kubernetes instance. But even for those, the set of `APIVersions` should be static unless you manage multiple clusters. And `argocd` actually just calls `helm template` instead of using the SDK so that's not even an actual example of use.

Reply:
I have an issue and PR raised that specifically tries to address this issue. I've run in data races  consuming the SDK concurrently https://github.com/helm/helm/issues/9618 https://github.com/helm/helm/pull/9619